lib/albacore/project.rb

```diff
@@ -210,6 +210,33 @@ def to_s
       path
     end
 
+    # Get Property folder path
+    def property_folder_path
+      read_property('AppDesignerFolder')
+    end
+
+    # Return path to properties folder
+    def properties_path
+      File.join @proj_path_base, property_folder_path
+    end
+
+    # Reads assembly version information
+    # @return string
+    def read_assembly_version
+      info= File.read(assembly_info)
+      v   = info.each_line
+                .select { |l| !(l.start_with?('//')||l.start_with?('/*')) && l.include?('AssemblyVersion') }.first
+      reg = /"(.*?)"/
+      reg.match(v).captures.first
+
+    end
+
+    # Get path to assemblyinfo.cs
+    # @return string
+    def assembly_info
```

Comment:
Another observation I have made is that vb projects do generate AssemblyInfo.vb files in visual studio. However, AppDesignerFolder is not present in vbproj files that are generated by the IDE. Furthermore, the path used for AssemblyInfo isn't the same either. (Leave it to MS to not make it consistent.) What do you think of implementing Project as an interface and making classes for the individual project types?

A switch would work here as well and be quicker to implement, but I wonder if that is the best option overall.

default_assembly_info does capture things better, especially since my local changes have it returning a default value of 1.0.0.0 now which is the default in generated AssemblyInfo files from VS.